File: computer3/Winter Term/9 - Mar 12/lvl3_hw_sol/sleeping_in_class.py
from sys import stdin
input = stdin.readline
for _ in range(int(input())):
    n = int(input())
    a = list(map(int, input().split()))
    vis = set()
    sum = 0
    for x in a:
        sum += x
        vis.add(sum)
    if sum == 0:
        print(0)
        continue
    ans = n+1
    for len in range(n, 0, -1):
        if sum % len != 0:
            continue
        val, flag = sum // len, True
        for x in range(val, sum+1, val):
            if x not in vis:
                flag = False
                break
        if flag:
            ans = n - len
            break
    print(ans)

File: computer3/Winter Term/9 - Mar 12/lvl3_hw_sol/sleeping_in_class.java
import java.util.*;
import java.io.*;
import java.time.*;
public class Main {
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	static PrintWriter pr = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out)));
	static StringTokenizer st;
	public static void main(String[] args) throws IOException {
		for(int T= readInt(); T>0; T--) {
			int n = readInt(), sum = 0; Set<Integer> vis=new HashSet();
			for(int i=0; i<n; i++) {
				sum += readInt();  vis.add(sum);
			}
			if(sum == 0) { System.out.println(0); continue; }
			int ans = Integer.MAX_VALUE;
			for(int len = n; len >= 1; len--) {
				if(sum % len != 0) continue;
				int val = sum / len; boolean flag = true;
				for(int x = val; x <= sum; x += val) {
					if(!vis.contains(x)) { flag =false; break; }
				}
				if(flag) {  ans = n-len; break; }
			}
			System.out.println(ans);
		}
	}
	static String next () throws IOException {
		while (st == null || !st.hasMoreTokens())
			st = new StringTokenizer(br.readLine().trim());
		return st.nextToken();
	}
	static long readLong () throws IOException {
		return Long.parseLong(next());
	}
	static int readInt () throws IOException {
		return Integer.parseInt(next());
	}
	static double readDouble () throws IOException {
		return Double.parseDouble(next());
	}
	static char readCharacter () throws IOException {
		return next().charAt(0);
	}
	static String readLine () throws IOException {
		return br.readLine().trim();
	}
}